client/test/router_store.spec.js

```diff
@@ -0,0 +1,669 @@
+    });
+    it('isDraft', () => {
+      assert.isFalse(
+        routerStore.isDraft(map(fixtures.message1)));
+      assert.isTrue(
+        routerStore.isDraft(map(fixtures.message4)));
+    });
+    it('getMailboxTotal', () => {
+      assert.equal(
+        routerStore.getMailboxTotal(),
+        fixtures.account.mailboxes[0].nbTotal
+      );
+      changeRoute({ });
+      assert.equal(
+        routerStore.getMailboxTotal(),
+        fixtures.account.mailboxes[0].nbTotal
+      );
+      changeRoute({ flags: MessageFilter.UNSEEN });
+      assert.equal(
+        routerStore.getMailboxTotal(),
+        fixtures.account.mailboxes[0].nbUnread
+      );
+      changeRoute({ flags: MessageFilter.FLAGGED });
+      assert.equal(
+        routerStore.getMailboxTotal(),
+        fixtures.account.mailboxes[0].nbFlagged
+      );
+    });
+    it('hasNextPage', () => {
+      dispatcher.dispatch({
+        type: ActionTypes.MESSAGE_FETCH_SUCCESS,
+        value: {
+          lastPage: {
+            info: 'last-page',
+            isComplete: false,
+          },
+          result: {},
+          timestamp: new Date(),
+        },
+      });
+      assert.isTrue(routerStore.hasNextPage());
+      dispatcher.dispatch({
+        type: ActionTypes.MESSAGE_FETCH_SUCCESS,
+        value: {
+          lastPage: {
+            info: 'last-page',
+            isComplete: true,
+          },
+          result: {},
+          timestamp: new Date(),
+        },
+      });
+      assert.isFalse(routerStore.hasNextPage());
+    });
+    it('getLastPage', () => {
+      loadMessages([fixtures.message1, fixtures.message2]);
+      assert.deepEqual(routerStore.getLastPage(), fixtures.lastPage);
+    });
+    it('isPageComplete', () => {
+      changeRoute({ });
+      loadMessages([fixtures.message1, fixtures.message2]);
+      routerStore.getMessagesList();
+      assert.isFalse(routerStore.isPageComplete());
+      const msgs = generateMessages();
+      loadMessages(msgs);
+      routerStore.getMessagesList();
+      assert.isTrue(routerStore.isPageComplete());
+    });
+    it('getMessagesList', () => {
+      const msgs = [];
+      for (let i = 0; i < UtilConstants.MSGBYPAGE + 3; i++) {
+        const message = _.clone(fixtures.message1);
+        message.id = `id${i}`;
+        message.messageID = `meid${i}`;
+        message.conversationID = `c${i}`;
+        msgs.push(message);
+      }
+      msgs[12].flags = [];
+      msgs[14].flags = [MessageFlags.FLAGGED];
+      msgs[15].flags = [MessageFlags.FLAGGED, MessageFlags.ATTACH];
+      changeRoute({ });
+      loadMessages(msgs);
+      assert.equal(routerStore.getMessagesList().size, msgs.length);
+      changeRoute({ flags: MessageFilter.UNSEEN });
+      assert.equal(routerStore.getMessagesList().size, 3);
+      changeRoute({ flags: MessageFilter.FLAGGED });
+      assert.equal(routerStore.getMessagesList().size, 2);
+      changeRoute({ flags: MessageFilter.ATTACH });
```

Comment:
You you use more blank lines to separate instruction blocks. The rule 1 blank lines to separate instruction blocks, and 2 between `classes`, `function`.